app/frontend/tests/unit/utils/permissionUtils.spec.js

```diff
@@ -0,0 +1,56 @@
+import { FormPermissions, IdentityProviders } from '@/utils/constants';
+import * as permissionUtils from '@/utils/permissionUtils';
+
+describe('checkFormSubmit', () => {
+  it('should be false when userForm is undefined', () => {
+    expect(permissionUtils.checkFormSubmit(undefined)).toBeFalsy();
+  });
+
+  it('should be false when idps is undefined', () => {
+    expect(permissionUtils.checkFormSubmit({ permissions: [] })).toBeFalsy();
+  });
+
+  it('should be false when permissions is undefined', () => {
+    expect(permissionUtils.checkFormSubmit({ idps: [] })).toBeFalsy();
+  });
+
+  it('should be true when idps is public', () => {
+    expect(permissionUtils.checkFormSubmit({ idps: [IdentityProviders.PUBLIC] })).toBeTruthy();
+  });
+
+  it('should be true when permissions is submission creator', () => {
+    expect(permissionUtils.checkFormSubmit({ permissions: [FormPermissions.SUBMISSION_CREATE] })).toBeTruthy();
+  });
+});
+
+describe('checkFormManage', () => {
+  it('should be false when userForm is undefined', () => {
+    expect(permissionUtils.checkFormManage(undefined)).toBeFalsy();
+  });
+
+  it('should be false when permissions is undefined', () => {
+    expect(permissionUtils.checkFormManage({})).toBeFalsy();
+  });
+
+  it('should be true when at least one appropriate permission exists', () => {
+    expect(permissionUtils.checkFormManage({ permissions: [FormPermissions.FORM_UPDATE] })).toBeTruthy();
+    expect(permissionUtils.checkFormManage({ permissions: [FormPermissions.FORM_DELETE] })).toBeTruthy();
+    expect(permissionUtils.checkFormManage({ permissions: [FormPermissions.DESIGN_UPDATE] })).toBeTruthy();
+    expect(permissionUtils.checkFormManage({ permissions: [FormPermissions.DESIGN_DELETE] })).toBeTruthy();
```

Comment:
just a thought..while the only link to 'team management' is on the form's 'Manage' page, I'm wondering if 
checkFormManage should also evaluate FormPermissions.TEAM_READ and FormPermissions.TEAM_UPDATE.
I guess that decision might affect this test.

Reply:
At least from a continuity perspective, the only way to reach team management is to go through the manage page first; in which case it would make sense to check if at least one of those permissions exists before allowing access to manage page. But as you mentioned, that will change intended behavior. Thoughts @loneil ?

Reply:
Reviewing this and other flows like it in 1643.